blog/2019-04-25-pulsar-debezium-tutorial.adoc

```diff
@@ -0,0 +1,242 @@
+Download https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.3.0/apache-pulsar-2.3.0-bin.tar.gz[Pulsar binary of 2.3.0 release] and https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.3.0/connectors/pulsar-io-kafka-connect-adaptor-2.3.0.nar[pulsar-io-kafka-connect-adaptor-2.3.0.nar of 2.3.0 release].
+In Pulsar, all Pulsar IO connectors are packaged as separate https://medium.com/hashmapinc/nifi-nar-files-explained-14113f7796fd[NAR] files.
+ 
+[source,bash]
+----
+$ wget https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.3.0/apache-pulsar-2.3.0-bin.tar.gz -O apache-pulsar-2.3.0-bin.tar.gz
+$ wget https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=pulsar/pulsar-2.3.0/connectors/pulsar-io-kafka-connect-adaptor-2.3.0.nar -O pulsar-io-kafka-connect-adaptor-2.3.0.nar
+$ tar zxf apache-pulsar-2.3.0-bin.tar.gz
+$ cd apache-pulsar-2.3.0
+$ mkdir connectors
+$ cp ../pulsar-io-kafka-connect-adaptor-2.3.0.nar connectors
+$ bin/pulsar standalone
+----
+
+++++
+<div class="imageblock centered-image">
+    <img src="/images/pulsar_tutorial/pulsar-mysql-1.png" style="max-width:100%; margin-bottom:20px; margin-top:20px;" class="responsive-image" alt="start pulsar standalone]">
+</div>
+++++
+
+### Step 3: Start Pulsar connector
+Start a Pulsar Debezium connector, with local run mode, in another terminal tab.
+The “debezium-mysql-source-config.yaml” file contains all the configuration, and main parameters are listed under the “configs” node. The .yaml file contains the "task.class" parameter, and Pulsar connector is a wrapper for the parameter. The configuration file also 
+includes MySQL related parameters (like server, port, user, password) and two names of Pulsar topics for "history" and "offset" storage.
+
+[source,bash]
+----
+$ bin/pulsar-admin source localrun  --sourceConfigFile debezium-mysql-source-config.yaml
+---- 
+
+The content in the “debezium-mysql-source-config.yaml” file is as follows.
+
+[source,bash]
+----
+tenant: "test"
+namespace: "test-namespace"
+name: "debezium-kafka-source"
+topicName: "kafka-connect-topic"
+archive: "connectors/pulsar-io-kafka-connect-adaptor-2.3.0.nar"
+
+parallelism: 1
+
+configs:
+  ## sourceTask
+  task.class: "io.debezium.connector.mysql.MySqlConnectorTask"
+
+  ## config for mysql, docker image: debezium/example-mysql:0.8
+  database.hostname: "localhost"
+  database.port: "3306"
+  database.user: "debezium"
+  database.password: "dbz"
+  database.server.id: "184054"
+  database.server.name: "dbserver1"
+  database.whitelist: "inventory"
+
+  database.history: "org.apache.pulsar.io.debezium.PulsarDatabaseHistory"
+  database.history.pulsar.topic: "history-topic"
+  database.history.pulsar.service.url: "pulsar://127.0.0.1:6650"
+  ## KEY_CONVERTER_CLASS_CONFIG, VALUE_CONVERTER_CLASS_CONFIG
+  key.converter: "org.apache.kafka.connect.json.JsonConverter"
```

Comment:
That's interesting; so you're actually re-using the Kafka converter types?

Reply:
right. Pulsar get record in Kafka format, here need re-using kafka converter to get data back.